single-function/lru-cache/submission.js

```diff
@@ -0,0 +1,66 @@
+class DoubleLinkedList {
+  constructor(key, value) {
+    this.key = key;
+    this.value = value;
+  }
+}
+
+class LRUCache {
+  constructor(capacity) {
+    this.capacity = capacity;
+    this.cache = {};
+    let head = new DoubleLinkedList(0, 0);
+    let tail = new DoubleLinkedList(0, 0);
+    head.next = tail;
+    tail.prev = head;
+    this.head = head;
+    this.tail = tail;
+    this.count = 0;
+  }
+
+  deleteDoubleLinkedList(node) {
+    node.prev.next = node.next;
+    node.next.prev = node.prev;
+  }
+
+  addNodeToHead(node) {
+    node.next = this.head.next;
+    this.head.next.prev = node;
+    this.head.next = node;
+    node.prev = this.head;
+  }
+
+  get(key) {
+    if (this.cache[key]) {
+      return this.ejectLinkedList(this.cache[key]).value;
+    } else {
+      return -1;
+    }
+  }
+
+  set(key, value) {
+    if (this.cache[key]) {
+      const node = this.ejectLinkedList(this.cache[key]);
```

Comment:
not sure what "eject" means here, at a glance -- it's ok to use more words in the name, would you mind renaming to be a little more self-describing about the purpose?